src/plugins/event_annotation/public/plugin.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Plugin, CoreSetup } from 'kibana/public';
+import { ExpressionsSetup } from '../../expressions/public';
+import { manualEventAnnotation, eventAnnotationGroup } from '../common';
+import { EventAnnotationService } from './event_annotation_service';
+
+interface SetupDependencies {
+  expressions: ExpressionsSetup;
+}
+
+/** @public */
+export type EventAnnotationPluginSetup = EventAnnotationService;
+
+/** @public */
+export type EventAnnotationPluginStart = EventAnnotationService;
+
+/** @public */
+export class EventAnnotationPlugin
+  implements Plugin<EventAnnotationPluginSetup, EventAnnotationPluginStart>
+{
+  private readonly eventAnnotationService = new EventAnnotationService();
+
+  public setup(core: CoreSetup, dependencies: SetupDependencies): EventAnnotationPluginSetup {
+    dependencies.expressions.registerFunction(manualEventAnnotation);
+    dependencies.expressions.registerFunction(eventAnnotationGroup);
+    return this.eventAnnotationService;
```

Comment:
nit: I see only one method in that service, not sure that it's a legal to execute this one during setup phase. I prefer to return {} or cleanup API

Reply:
But at some point I'll add extra functionality here. I will not correct it in this PR but revisit later to make sure if it doesn't break any rules.